host.py

```diff
@@ -375,7 +376,10 @@ def read_output(cmd: str, log_level: int) -> Result:
                 return read_output(cmd, log_level)
             except Exception as e:
                 logger.log(log_level, e)
-                logger.log(log_level, f"Connection lost while running command {cmd}, reconnecting...")
+                cmd_str = ""
+                if not quiet:
```

Comment:
Shouldn't we always log "cmd"?  "cmd_str" is empty here.

Reply:
I think this is a bit fragile. Since if we want these logs to be public facing. There could be a potential for someone to accidentally check in something and forget to set quiet=true.

Maybe we need a cmd_str_santized where when we access secrets we obfuscate the secrets with e.g. "*****".

Reply:
I can't think of a way in python to handle this where we would still be able to access the value of the string in run().

We could add a class to override the __str__ method which would prevent the string from being able to be accidentally printed to stdout, i.e.

```
class Secret:
    def __init__(self, value):
        self._value = value
    
    def __str__(self):
        return '******'
    
    def __repr__(self):
        return '******'
    
    def get_value(self):
        return self._value
```

But when we add this string to a cmd to pass to run(), it would be treated as "*****" rather than the value of the string...there must be some way to do this, but it I am not sure how.

Reply:
Hmm I would have tried it your way like you did with secret.

Reply:
One dirty way is to wrap passwords with some header that can be regex out. When we see this regex in the logger then we will replace it with ****** . When we see this regex in the cmd class then we will remove the header before running the command.

Although dirty this would also require the developer to make sure they use the class which puts the header in front of the string.

I am not sure if this is a good solution to this problem or not.